addons/sourcemod/scripting/vsh/abilities/ability_rage_hop.sp

```diff
@@ -0,0 +1,184 @@
+static int g_iRageHopMaxCharge[TF_MAXPLAYERS+1];
+static float g_flRageHopMaxHeight[TF_MAXPLAYERS+1];
+static float g_flRageHopMaxDistance[TF_MAXPLAYERS+1];
+static float g_flBombDamage[TF_MAXPLAYERS+1];
+static float g_flBombRadius[TF_MAXPLAYERS+1];
+static float g_flFloatEndTime[TF_MAXPLAYERS+1];
+static bool g_bStompEnabled[TF_MAXPLAYERS+1];
+static float g_flDuration[TF_MAXPLAYERS+1];
+static float g_flDurationSuper[TF_MAXPLAYERS+1];
+
+methodmap CRageHop < SaxtonHaleBase
+{
+	property float flDuration
+	{
+		public get()
+		{
+			return g_flDuration[this.iClient];
+		}
+		public set(float val)
+		{
+			g_flDuration[this.iClient] = val;
+		}
+	}
+	
+	property float flDurationSuper
+	{
+		public get()
+		{
+			return g_flDurationSuper[this.iClient];
+		}
+		public set(float val)
+		{
+			g_flDurationSuper[this.iClient] = val;
+		}
+	}
+	
+	property int iMaxJumpCharge
+	{
+		public get()
+		{
+			return g_iRageHopMaxCharge[this.iClient];
+		}
+		public set(int val)
+		{
+			g_iRageHopMaxCharge[this.iClient] = val;
+		}
+	}
+	
+	property float flMaxHeight
+	{
+		public get()
+		{
+			return g_flRageHopMaxHeight[this.iClient];
+		}
+		public set(float val)
+		{
+			g_flRageHopMaxHeight[this.iClient] = val;
+		}
+	}
+	
+	property float flMaxDistance
+	{
+		public get()
+		{
+			return g_flRageHopMaxDistance[this.iClient];
+		}
+		public set(float val)
+		{
+			g_flRageHopMaxDistance[this.iClient] = val;
+		}
+	}
+	
+	property float flBombDamage
+	{
+		public get()
+		{
+			return g_flBombDamage[this.iClient];
+		}
+		public set(float val)
+		{
+			g_flBombDamage[this.iClient] = val;
+		}
+	}
+	
+		property float flBombRadius
+	{
+		public get()
+		{
+			return g_flBombRadius[this.iClient];
+		}
+		public set(float val)
+		{
+			g_flBombRadius[this.iClient] = val;
+		}
+	}
+	
+	public CRageHop(CRageHop ability)
+	{
+		//Default values, these can be changed if needed
+		ability.iMaxJumpCharge = 1200;
+		ability.flMaxHeight = 600.0;
+		ability.flMaxDistance = 4.5;
+		
+		ability.flBombDamage = 10.0;
+		ability.flBombRadius = 200.0;
+		
+		ability.flDuration = 45.0;
+		ability.flDurationSuper = 16.0;
+		
+	}
+	
+	public void OnThink()
+	{
+		if (GameRules_GetRoundState() == RoundState_Preround) return;
+		if (!IsPlayerAlive(this.iClient)) return;
+		
+		int iTeam = GetClientTeam(this.iClient);
+		
+		if(g_bStompEnabled[this.iClient])
+		{
+			float vecExplosionPos[3], vecExplosionOrigin[3];
```

Comment:
``vecExplosionPos`` is pointless, only has one use in ``TF2_Explode`` below and is assigned ``vecExplosionOrigin`` anyway